### PR TITLE
Refactor: Add focus styling to Fronts Cards

### LINKED ISSUE #5177: Focus styling for CardLinks
In the onwards carousel-type component on Fronts, there is no visible focus styling for the article cards, which makes keyboard navigation difficult. At the same time, using the keyboard one is able to highlight and click through to a `#comments` link for an article, even if comments are not enabled and there's no visible link. (Screenshot below is of the comments link being focused -- small blue square in the bottom right corner of the middle card.)

Image

- [x] Visible focus highlights should be added to the cards (existing hover styling is not very pronounced, so might be better to use something else, e.g. the focus ring from Source?).
- [x] #5634

### PR DESCRIPTION
**Pull request by bryophyta**
## What does this change?

- Add focusHalo to CardLink on focus.
- Remove unneeded styling from CardLink:
  - Styling of CardLink itself was redundant because opacity was set to 0.
  - The styling applied indirectly to `.image-overlay` was redundant because it
  didn't actually select anything (perhaps copied mistakenly from CardWrapper?)

## Why?

There was no visible focus indicator on these cards before. Closes #5177 

## Queries
I couldn't see what the code that I've deleted from `CardLink.tsx` was doing, but that doesn't mean that it wasn't doing anything! If anyone has more information, please let me know!

## Screenshots
Before (no visible styling when focused):
image

After: 
image

### CODE REVIEW

**Comment by oliverlloyd** [comment]
Ha, great spot! Looks like we had the same hover state code set in two places, here and - like you say - in `CardWrapper`. I have no idea how that came about but well done for digging into this and fixing it

File: dotcom-rendering/src/web/components/Card/components/CardLink.tsx [hunk 1]
```diff
@@ -13,104 +10,46 @@ const fauxLinkStyles = css`
...
-		:before {
-			content: '';
-			position: absolute;
-			left: 0;
-			top: 0;
-			right: 0;
-			bottom: 0;
-		}
-
-		:hover .image-overlay {
```
Comment (bryophyta) [inline dotcom-rendering/src/web/components/Card/components/CardLink.tsx:42]: this rule is also present in `CardWrapper`, and I couldn't see any scenario where there would be an element with `.image-ovoerlay` inside the `CardLink`, though I may have missed something?

**Comment by bryophyta** [comment]
Thanks @oliverlloyd ! I've looked over it again and  removed a bit more css. I can't see any ways in which this affects the final rendering, but I'm not 100% sure -- would you mind casting another eye over the most recent commit at some point, please?

**Review by oliverlloyd — APPROVED** [review]
I checked this out locally and can verify that the hover states for both the whole card overlay and also the sublinks work as expected 👍 

Nice pruning work

**Comment by bryophyta** [comment]
> Nice pruning work

Thanks! 🧑‍🌾 (TIL that there's aren't many good gardening emojis 😔 )

I'll merge tomorrow 👍

**Pull request merged**